File: src/main/java/qmech/lib/objects/equip/ArmorBase.java
package qmech.lib.objects.equip;

import java.util.List;

import cpw.mods.fml.common.registry.GameRegistry;
import net.minecraft.entity.Entity;
import net.minecraft.entity.player.EntityPlayer;
import qmech.lib.objects.CreativeTabBase;
import qmech.lib.util.RecipeHelper;
import qmech.mod.Reference;
import net.minecraft.item.ItemArmor;
import net.minecraft.item.ItemStack;

public class ArmorBase extends ItemArmor {

    public static String[] suffixes = new String[]
            {
                    "helmet", "chestplate", "leggings", "boots"
            };

    public int armorPiece;

	public ArmorBase(ArmorTypeBase armorType, int armorIndex, CreativeTabBase ctab) {
		super(armorType.getArmorMaterial(), 0, armorIndex);
        String intName = String.format("%s_%s", armorType.getArmorName(), suffixes[armorIndex]);
        this.setTextureName(Reference.MOD_ID + ":" + intName);
        this.setInternalName(intName);
        this.setCreativeTab(ctab);

        armorPiece = armorIndex;

        RecipeHelper.shapedRecipe(new ItemStack(this), ArmorTypeBase.craftingRecipes.get(armorIndex), 'x', armorType.getMaterial());

        GameRegistry.registerItem(this, intName);
	}
    
    public ArmorBase setInternalName (String name) {
        this.setUnlocalizedName(name);
        return this;
    }

    @Override
    public void addInformation(ItemStack itemstack, EntityPlayer entityplayer, List list, boolean flag)
    {
        list.add("HP: " + (itemstack.getMaxDamage() - itemstack.getItemDamage()));
    }
    
    public void registerItem () { 
        GameRegistry.registerItem(this, Reference.MOD_ID+":"+this.getUnlocalizedName().substring(5));
    }

    public String getArmorTexture(ItemStack stack, Entity entity, int slot, String type) {
        if (stack.getItem() instanceof ArmorBase) {
            ArmorBase armor = (ArmorBase) stack.getItem();
            int layer = 1;
            if (armor.armorPiece == 2) {
                layer = 2;
            }
            return String.format("%s:textures/models/armor/%s_%s.png", Reference.MOD_ID, armor.getArmorMaterial().name(), layer);
        }
        return null;
    }

}

File: src/main/scala/geek/lawsof/physics/lib/objects/equip/ArmorBase.scala
package geek.lawsof.physics.lib.objects.equip

import net.minecraft.item.{ItemStack, ItemArmor}
import net.minecraft.entity.player.EntityPlayer
import java.util
import net.minecraft.entity.Entity
import geek.lawsof.physics.lib.objects.equip.types.ArmorTypeBase
import geek.lawsof.physics.lib.objects.CreativeTabBase
import cpw.mods.fml.common.registry.GameRegistry
import geek.lawsof.physics.Reference

/**
 * Created by anshuman on 28-05-2014.
 */
object ArmorBase {
  val suffixes = Array("helmet", "chestplate", "leggings", "boots")
}

class ArmorBase(val armorMaterial: ArmorTypeBase, val armorPart: Int, ctab: CreativeTabBase) extends ItemArmor(armorMaterial.armorType, 0, armorPart) {
  val intName = s"${armorMaterial.armorName}_${ArmorBase.suffixes(armorPart)}"
  setTextureName(s"${Reference.MOD_ID}:$intName")
  setUnlocalizedName(intName)
  setCreativeTab(ctab)

  //  Recipes.shapedRecipe(new ItemStack(this), ArmorTypeBase.craftingRecipes(armorPart), 'x', armorMaterial.craftMaterial)

  GameRegistry.registerItem(this, intName)

  override def addInformation(stack: ItemStack, par2EntityPlayer: EntityPlayer, par3List: util.List[_], par4: Boolean) = {
    var list = par3List.asInstanceOf[util.List[String]]
    list.add(s"HP : ${stack.getMaxDamage - stack.getItemDamage}/${stack.getMaxDamage}")
    list.add(s"Protection : ${armorMaterial.reductionAmounts(armorPart)}")
  }

  override def getArmorTexture(stack: ItemStack, entity: Entity, slot: Int, `type`: String): String = {
    val layer = slot match {
      case 2 => "_2"
      case _ => "_1"
    }
    val armorName = stack.getItem.asInstanceOf[ArmorBase].getArmorMaterial.name()
    //    val armorName = this.getArmorMaterial.name()
    s"${Reference.MOD_ID}:/textures/models/armor/$armorName$layer.png"
  }

  def newItemStack (amt: Int = 1) = new ItemStack(this, amt)
}
